issue #2528 - change interpretation of lt and gt for date searches

lmsurpre opened the pull request:
We use the range interpretation of the search prefixes for both date and
number/quantity searches.  The spec says:
* `lt` means "the range below the search value intersects (i.e. overlaps) with the
range of the target value"
* `gt` means "the range above the search value intersects (i.e. overlaps) with the
range of the target value"

For numbers, the spec additionally says:
* When a comparison prefix in the set gt, lt, ge, le, sa & eb is provided, the implicit precision of the number is ignored, and they are treated as if they have arbitrarily high precision

Previously, we interpreted date search in the same way and so a search like `gt2016`
would return any value above `2016-01-01 00:00:00` in the server's
timezone.  However, we were not consistent with this because, for some reason,
we decided that `lt2016` should return any value below `2017-01-01 00:00:00`.

However, even if we properly implemented this "arbitrarily high precision" clause for date search,
this interpretation leads to confusing and unintuitive results like `gt2016` returning dates in 2016.
Instead, we will now interpret the "search value" of 2016 to be the
range `[2016-01-01 00:00:00, 2017-01-01 00:00:00)` and therefor a search for `gt2016` will
only find target values above 2017-01-01 00:00:00 (inclusive).

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

prb112 commented:
Should the new  builder util be updated with this change?

tbieste commented on fhir-persistence/src/test/resources/logging.unitTest.properties:
```diff
@@ -49,5 +49,6 @@ java.util.logging.ConsoleHandler.formatter = java.util.logging.SimpleFormatter
 ############################################################
```
Update copyright date

lmsurpre replied:
lol, this one has a beginning copyright of 2000... should we change that to 2016 when our project started?

tbieste replied:
Sounds good to me.

JohnTimm reviewed (approved):
LGTM

lmsurpre commented:
> Should the new builder util be updated with this change?

I reworked the changes so that its now contained in the persistence layer and I updated both the old and new behavior classes
